onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -324,6 +330,93 @@ const std::vector<std::string>& GetAvailableProviders() {
   return available_providers;
 }
 
+#ifdef USE_CUDA
+bool IsCudaDeviceIdValid(int id) {
+    
+    int num_devices = 0;
+    CUDA_CALL_THROW(cudaGetDeviceCount(&num_devices));
+
+    if (0 == num_devices)
+    {
+        printf("your system does not have a CUDA capable device.\n");
```

Comment:
don't use printf.
use default logger for instances where logger isn't available.
e.g. 
```
LOGS_DEFAULT(WARNING) << "message"
```